main.nf

```diff
@@ -807,7 +871,7 @@ if (merge) {
   }.map {
     [it[0], (it[2].sum()/it[2].size()).toInteger(), it[3][0], it[4]]
   }.set {
-    maps_to_merge
+    maps_to_merge_assembly
   }
 
   // Check differing read lengths and directions
```

Comment:
Will have to do this check for both merge options.

Reply:
I meant the check for differing read lengths and directions. Within a group all reads must have same lengths and direction (RF/FR). This is necessary for assembly (same direction) and quantification (same direction and length). So we need something like:

    .groupTuple().tap {
        maps_to_check_before_merge_for_assembly
    }

    
    .groupTuple().tap {
        maps_to_check_before_merge_for_quantification
    }


`maps_to_check_before_merge_for_assembly` and `maps_to_check_before_merge_for_quantification` will be channels of lists:

    [
        name of group1,
        [length of map1, length of map2, ...],
        [direction of map1, direction of map2, ...], 
        [path of map1, path of map2, ...]
    ],
    [
        name of group2,
        [length of map3, length of map4, ...],
        [direction of map3, direction of map4, ...], 
        [path of map3, path of map4, ...]
    ],
    ...